src/gui/macosshiftclickhandler.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2025  Luke Memet <lukemmtt@users.noreply.github.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+#include <QObject>
+#include <QPersistentModelIndex>
+
+class TransferListWidget;
+
+// Workaround for QTBUG-115838: Shift-click range selection not working properly on macOS
+class MacOSShiftClickHandler final : public QObject
+{
+    Q_OBJECT
+    Q_DISABLE_COPY_MOVE(MacOSShiftClickHandler)
+
+public:
+    explicit MacOSShiftClickHandler(TransferListWidget *treeView);
+
+private:
+    bool eventFilter(QObject *watched, QEvent *event) override;
+
+    TransferListWidget *m_treeView;
```

Comment:
```suggestion
    TransferListWidget *m_treeView = nullptr;
```

Reply:
Also just `QTreeView` should suffice:
```suggestion
    QTreeView *m_treeView = nullptr;
```

Reply:
@lukemmtt 
Just a reminder to make sure you have seen the second suggestion in this thread:
https://github.com/qbittorrent/qBittorrent/pull/22284#discussion_r1957188179

Reply:
Thanks! 🙏🏻